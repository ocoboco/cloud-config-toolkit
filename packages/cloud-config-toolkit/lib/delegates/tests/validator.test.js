const ValidatorDelegate = require('../validator');

describe('ValidatorDelegate', function () {
  describe('validate()', function() {
    test('returns no errors when valid', function () {
      const delegate = new ValidatorDelegate({
        validator: {
          isValid() {
            return true;
          },
          getErrors() {
            return [];
          }
        }
      });
      expect(delegate.validate()).toMatchSnapshot();
    });

    test('returns errors when invalid', function () {
      const delegate = new ValidatorDelegate({
        validator: {
          isValid() {
            return false;
          },
          getErrors() {
            return [
              { message: 'Oops, something went wrong' }
            ];
          }
        }
      });
      expect(delegate.validate()).toMatchSnapshot();
    });
  });
});