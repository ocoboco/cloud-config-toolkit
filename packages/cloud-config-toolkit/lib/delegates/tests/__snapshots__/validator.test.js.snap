// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ValidatorDelegate validate() returns errors when invalid 1`] = `
Object {
  "errors": Array [
    Object {
      "message": "Oops, something went wrong",
    },
  ],
  "isValid": false,
}
`;

exports[`ValidatorDelegate validate() returns no errors when valid 1`] = `
Object {
  "errors": Array [],
  "isValid": true,
}
`;
